Various warnings when building with GCC 12

When building CMSIS-DSP with GCC 12.1, `Wmaybe-uninitialized` warnings are seen:


```
In function '__arm_vqmovnbq_s16',
    inlined from 'arm_float_to_q7' at /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:104:18:
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.1.0/include/arm_mve.h:6384:10: error: 'vecDst' may be used uninitialized [-Werror=maybe-uninitialized]
 6384 |   return __builtin_mve_vqmovnbq_sv8hi (__a, __b);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c: In function 'arm_float_to_q7':
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:71:21: note: 'vecDst' was declared here
   71 |     q7x16_t         vecDst;
      |                     ^~~~~~
In function '__arm_vqmovnbq_s32',
    inlined from 'arm_float_to_q7' at /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:99:18:
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.1.0/include/arm_mve.h:6587:10: error: 'oddVec' may be used uninitialized [-Werror=maybe-uninitialized]
 6587 |   return __builtin_mve_vqmovnbq_sv4si (__a, __b);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c: In function 'arm_float_to_q7':
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:70:28: note: 'oddVec' was declared here
   70 |     q15x8_t         evVec, oddVec;
      |                            ^~~~~~
In function '__arm_vqmovnbq_s32',
    inlined from 'arm_float_to_q7' at /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:94:17:
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.1.0/include/arm_mve.h:6587:10: error: 'evVec' may be used uninitialized [-Werror=maybe-uninitialized]
 6587 |   return __builtin_mve_vqmovnbq_sv4si (__a, __b);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c: In function 'arm_float_to_q7':
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q7.c:70:21: note: 'evVec' was declared here
   70 |     q15x8_t         evVec, oddVec;
      |                     ^~~~~
cc1: all warnings being treated as errors
[298/374] Building C object modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_q15_to_q7.c.obj
FAILED: modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_q15_to_q7.c.obj 
ccache /home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DARM_ALL_FAST_TABLES -DARM_DSP_CONFIG_TABLES -DARM_FAST_ALLOW_TABLES -DARM_FFT_ALLOW_TABLES -DARM_TABLE_RECIP_Q15 -DARM_TABLE_RECIP_Q31 -DARM_TABLE_SIN_F32 -DARM_TABLE_SIN_Q15 -DARM_TABLE_SIN_Q31 -DKERNEL -DTC_RUNID=5daaff6597b061944ae4b8f0da905274 -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/PrivateInclude -I/home/gmarull/ws/nordic/zephyrproject/zephyr/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/twister-out/mps3_an547/tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.misc.fpu/zephyr/include/generated -I/home/gmarull/ws/nordic/zephyrproject/zephyr/soc/arm/arm/mps3 -I/home/gmarull/ws/nordic/zephyrproject/zephyr/lib/libc/newlib/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/subsys/testsuite/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/subsys/testsuite/ztest/include -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include -fno-strict-aliasing -Os -imacros /home/gmarull/ws/nordic/zephyrproject/zephyr/twister-out/mps3_an547/tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.misc.fpu/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m55 -mthumb -mabi=aapcs -mfpu=auto -mfloat-abi=hard -mfp16-format=ieee --sysroot=/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/gmarull/ws/nordic/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject/zephyr/tests/lib/cmsis_dsp/filtering=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mcmse -specs=nano.specs -std=c99 -MD -MT modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_q15_to_q7.c.obj -MF modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_q15_to_q7.c.obj.d -o modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_q15_to_q7.c.obj -c /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_q15_to_q7.c
In file included from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include/arm_math_types.h:249,
                 from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include/dsp/support_functions.h:30,
                 from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_q15_to_q7.c:29:
In function '__arm_vqshrnbq_n_s16',
    inlined from 'arm_q15_to_q7' at /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_q15_to_q7.c:74:18:
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.1.0/include/arm_mve.h:8812:10: error: 'vecDst' may be used uninitialized [-Werror=maybe-uninitialized]
 8812 |   return __builtin_mve_vqshrnbq_n_sv8hi (__a, __b, __imm);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_q15_to_q7.c: In function 'arm_q15_to_q7':
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_q15_to_q7.c:63:13: note: 'vecDst' was declared here
   63 |     q7x16_t vecDst;
      |             ^~~~~~
cc1: all warnings being treated as errors
[299/374] Building C object modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_float_to_q15.c.obj
FAILED: modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_float_to_q15.c.obj 
ccache /home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DARM_ALL_FAST_TABLES -DARM_DSP_CONFIG_TABLES -DARM_FAST_ALLOW_TABLES -DARM_FFT_ALLOW_TABLES -DARM_TABLE_RECIP_Q15 -DARM_TABLE_RECIP_Q31 -DARM_TABLE_SIN_F32 -DARM_TABLE_SIN_Q15 -DARM_TABLE_SIN_Q31 -DKERNEL -DTC_RUNID=5daaff6597b061944ae4b8f0da905274 -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/PrivateInclude -I/home/gmarull/ws/nordic/zephyrproject/zephyr/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/twister-out/mps3_an547/tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.misc.fpu/zephyr/include/generated -I/home/gmarull/ws/nordic/zephyrproject/zephyr/soc/arm/arm/mps3 -I/home/gmarull/ws/nordic/zephyrproject/zephyr/lib/libc/newlib/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/subsys/testsuite/include -I/home/gmarull/ws/nordic/zephyrproject/zephyr/subsys/testsuite/ztest/include -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include -fno-strict-aliasing -Os -imacros /home/gmarull/ws/nordic/zephyrproject/zephyr/twister-out/mps3_an547/tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.misc.fpu/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m55 -mthumb -mabi=aapcs -mfpu=auto -mfloat-abi=hard -mfp16-format=ieee --sysroot=/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/gmarull/ws/nordic/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject/zephyr/tests/lib/cmsis_dsp/filtering=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gmarull/ws/nordic/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mcmse -specs=nano.specs -std=c99 -MD -MT modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_float_to_q15.c.obj -MF modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_float_to_q15.c.obj.d -o modules/cmsis/CMSIS/DSP/Source/SupportFunctions/CMakeFiles/..__modules__hal__cmsis__CMSIS__DSP__Source__SupportFunctions.dir/arm_float_to_q15.c.obj -c /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q15.c
In file included from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include/arm_math_types.h:249,
                 from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Include/dsp/support_functions.h:30,
                 from /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q15.c:29:
In function '__arm_vqmovnbq_s32',
    inlined from 'arm_float_to_q15' at /home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q15.c:87:18:
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.1.0/include/arm_mve.h:6587:10: error: 'vecDst' may be used uninitialized [-Werror=maybe-uninitialized]
 6587 |   return __builtin_mve_vqmovnbq_sv4si (__a, __b);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q15.c: In function 'arm_float_to_q15':
/home/gmarull/ws/nordic/zephyrproject/modules/hal/cmsis/CMSIS/DSP/Source/SupportFunctions/arm_float_to_q15.c:71:21: note: 'vecDst' was declared here
   71 |     q15x8_t         vecDst;
      |                     ^~~~~~
cc1: all warnings being treated as errors
```

--- a/Source/MatrixFunctions/arm_mat_add_f16.c
+++ b/Source/MatrixFunctions/arm_mat_add_f16.c
@@ -62,7 +62,7 @@ arm_status arm_mat_add_f16(
     arm_status status;  
     uint32_t  numSamples;       /* total number of elements in the matrix  */
     float16_t *pDataA, *pDataB, *pDataDst;
-    f16x8_t vecA, vecB, vecDst;
+    f16x8_t vecA, vecB, vecDst = { 0 };
     float16_t const *pSrcAVec;
     float16_t const *pSrcBVec;
     uint32_t  blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_add_f32.c
+++ b/Source/MatrixFunctions/arm_mat_add_f32.c
@@ -88,7 +88,7 @@ arm_status arm_mat_add_f32(
     arm_status status;  
     uint32_t  numSamples;       /* total number of elements in the matrix  */
     float32_t *pDataA, *pDataB, *pDataDst;
-    f32x4_t vecA, vecB, vecDst;
+    f32x4_t vecA, vecB, vecDst = { 0 };
     float32_t const *pSrcAVec;
     float32_t const *pSrcBVec;
     uint32_t  blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_add_q15.c
+++ b/Source/MatrixFunctions/arm_mat_add_q15.c
@@ -59,7 +59,7 @@ arm_status arm_mat_add_q15(
 {
     uint32_t        numSamples;       /* total number of elements in the matrix  */
     q15_t          *pDataA, *pDataB, *pDataDst;
-    q15x8_t       vecA, vecB, vecDst;
+    q15x8_t       vecA, vecB, vecDst = { 0 };
     q15_t const   *pSrcAVec;
     q15_t const   *pSrcBVec;
     uint32_t        blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_add_q31.c
+++ b/Source/MatrixFunctions/arm_mat_add_q31.c
@@ -59,7 +59,7 @@ arm_status arm_mat_add_q31(
     arm_status status;                             /* status of matrix addition */
     uint32_t        numSamples;       /* total number of elements in the matrix  */
     q31_t          *pDataA, *pDataB, *pDataDst;
-    q31x4_t       vecA, vecB, vecDst;
+    q31x4_t       vecA, vecB, vecDst = { 0 };
     q31_t const   *pSrcAVec;
     q31_t const   *pSrcBVec;
     uint32_t        blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_sub_f16.c
+++ b/Source/MatrixFunctions/arm_mat_sub_f16.c
@@ -60,7 +60,7 @@ arm_status arm_mat_sub_f16(
     arm_status status;                             /* status of matrix subtraction */
     uint32_t  numSamples;       /* total number of elements in the matrix  */
     float16_t *pDataA, *pDataB, *pDataDst;
-    f16x8_t vecA, vecB, vecDst;
+    f16x8_t vecA, vecB, vecDst = { 0 };
     float16_t const *pSrcAVec;
     float16_t const *pSrcBVec;
     uint32_t  blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_sub_f32.c
+++ b/Source/MatrixFunctions/arm_mat_sub_f32.c
@@ -85,7 +85,7 @@ arm_status arm_mat_sub_f32(
     arm_status status;                             /* status of matrix subtraction */
     uint32_t  numSamples;       /* total number of elements in the matrix  */
     float32_t *pDataA, *pDataB, *pDataDst;
-    f32x4_t vecA, vecB, vecDst;
+    f32x4_t vecA, vecB, vecDst = { 0 };
     float32_t const *pSrcAVec;
     float32_t const *pSrcBVec;
     uint32_t  blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_sub_q15.c
+++ b/Source/MatrixFunctions/arm_mat_sub_q15.c
@@ -59,7 +59,7 @@ arm_status arm_mat_sub_q15(
 {
     uint32_t        numSamples;       /* total number of elements in the matrix  */
     q15_t          *pDataA, *pDataB, *pDataDst;
-    q15x8_t       vecA, vecB, vecDst;
+    q15x8_t       vecA, vecB, vecDst = { 0 };
     q15_t const   *pSrcAVec;
     q15_t const   *pSrcBVec;
     uint32_t        blkCnt;           /* loop counters */

--- a/Source/MatrixFunctions/arm_mat_sub_q31.c
+++ b/Source/MatrixFunctions/arm_mat_sub_q31.c
@@ -58,7 +58,7 @@ arm_status arm_mat_sub_q31(
 {
     uint32_t        numSamples;       /* total number of elements in the matrix  */
     q31_t          *pDataA, *pDataB, *pDataDst;
-    q31x4_t       vecA, vecB, vecDst;
+    q31x4_t       vecA, vecB, vecDst = { 0 };
     q31_t const   *pSrcAVec;
     q31_t const   *pSrcBVec;
     uint32_t        blkCnt;           /* loop counters */

--- a/Source/SupportFunctions/arm_float_to_q15.c
+++ b/Source/SupportFunctions/arm_float_to_q15.c
@@ -68,7 +68,7 @@ void arm_float_to_q15(
     uint32_t         blkCnt;
     float32_t       maxQ = (float32_t) Q15_MAX;
     f32x4x2_t       tmp;
-    q15x8_t         vecDst;
+    q15x8_t         vecDst = { 0 };
 #ifdef ARM_MATH_ROUNDING
     float32_t in;
 #endif

--- a/Source/SupportFunctions/arm_float_to_q7.c
+++ b/Source/SupportFunctions/arm_float_to_q7.c
@@ -67,8 +67,8 @@ void arm_float_to_q7(
     uint32_t         blkCnt;     /* loop counters */
     float32_t       maxQ = powf(2.0, 7);
     f32x4x4_t       tmp;
-    q15x8_t         evVec, oddVec;
-    q7x16_t         vecDst;
+    q15x8_t         evVec = { 0 }, oddVec = { 0 };
+    q7x16_t         vecDst = { 0 };
     float32_t const *pSrcVec;
 #ifdef ARM_MATH_ROUNDING
     float32_t in;

--- a/Source/SupportFunctions/arm_q15_to_q7.c
+++ b/Source/SupportFunctions/arm_q15_to_q7.c
@@ -60,7 +60,7 @@ void arm_q15_to_q7(
     uint32_t  blkCnt;           /* loop counters */
     q15x8x2_t tmp;
     q15_t const *pSrcVec;
-    q7x16_t vecDst;
+    q7x16_t vecDst = { 0 };
 
 
     pSrcVec = (q15_t const *) pSrc;

--- a/Source/SupportFunctions/arm_q31_to_q15.c
+++ b/Source/SupportFunctions/arm_q31_to_q15.c
@@ -58,7 +58,7 @@ void arm_q31_to_q15(
 {
     uint32_t  blkCnt;           /* loop counters */
     q31x4x2_t tmp;
-    q15x8_t vecDst;
+    q15x8_t vecDst = { 0 };
     q31_t const *pSrcVec;
 
 

--- a/Source/SupportFunctions/arm_q31_to_q7.c
+++ b/Source/SupportFunctions/arm_q31_to_q7.c
@@ -58,8 +58,8 @@ void arm_q31_to_q7(
 {
     uint32_t  blkCnt;           /* loop counters */
     q31x4x4_t tmp;
-    q15x8_t evVec, oddVec;
-    q7x16_t  vecDst;
+    q15x8_t evVec = { 0 }, oddVec = { 0 };
+    q7x16_t  vecDst = { 0 };
     q31_t const *pSrcVec;
 
     pSrcVec = (q31_t const *) pSrc;

